src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -668,18 +700,18 @@ private Compilation CreateEmptyCompilation()
 
                         compilationWithoutGeneratedDocuments = await action.TransformCompilationAsync(compilationWithoutGeneratedDocuments, cancellationToken).ConfigureAwait(false);
 
-                        if (compilationWithGeneratedDocuments != null)
+                        if (staleCompilationWithGeneratedDocuments != null)
                         {
                             // Also transform the compilation that has generated files; we won't do that though if the transformation either would cause problems with
                             // the generated documents, or if don't have any source generators in the first place.
                             if (action.CanUpdateCompilationWithStaleGeneratedTreesIfGeneratorsGiveSameOutput &&
                                 oldState.SourceGenerators.Any())
                             {
-                                compilationWithGeneratedDocuments = await action.TransformCompilationAsync(compilationWithGeneratedDocuments, cancellationToken).ConfigureAwait(false);
+                                staleCompilationWithGeneratedDocuments = await action.TransformCompilationAsync(staleCompilationWithGeneratedDocuments, cancellationToken).ConfigureAwait(false);
                             }
                             else
                             {
-                                compilationWithGeneratedDocuments = null;
+                                staleCompilationWithGeneratedDocuments = null;
                             }
```

Comment:
i don't really understnad this logic, and how staleCompilation is used.  Need @jasonmalinowski thoughts that everything stays sound, even with the one place where i'm now possibly passing a different value from before, and the new place where we could consider doing hte same.